lib/Doctrine/DBAL/Schema/Comparator.php

```diff
@@ -421,7 +422,11 @@ public function diffColumn(Column $column1, Column $column2)
 
         $changedProperties = [];
 
-        foreach (['type', 'notnull', 'unsigned', 'autoincrement'] as $property) {
+        if (get_class($properties1['type']) !== get_class($properties2['type'])) {
```

Comment:
How does it work with non-object types? Isn't it supposed to throw some warning?
```sh
$ php -r "var_dump(get_class('string'));"

Warning: get_class() expects parameter 1 to be object, string given in Command line code on line 1

Call Stack:
    0.0001     466744   1. {main}() Command line code:0

Command line code:1:
bool(false)
```
Also, will it work correctly with two different non-object types?
```sh
$ php -r "var_dump(get_class('string') === get_class('integer'));"

Warning: get_class() expects parameter 1 to be object, string given in Command line code on line 1

Call Stack:
    0.0001     466880   1. {main}() Command line code:0


Warning: get_class() expects parameter 1 to be object, string given in Command line code on line 1

Call Stack:
    0.0001     466880   1. {main}() Command line code:0

Command line code:1:
bool(true)
```

Reply:
I see. A type is always an object but it's a nullable object.